js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,78 @@
+describe("Enhanced date", function () {
+	
+
+	describe("getDayName", function () {
+		before(function() {
+			enhancedDate.setDate(new Date("05-23-2015"));
+		});
+		it("should return Saturday", function() {
+			expect(enhancedDate.getDayName()).to.be.equal("Saturday");
+		});
+		it("should be a string", function() {
+			expect(enhancedDate.getDayName()).to.be.a("string");
+		});
+	});
+
+	describe("setDate", function() {		
+
+		it("should return instanceof date", function () {
+        	expect(enhancedDate.setDate(new Date("05-23-2015"))).to.be.an.instanceof(Date);        	
+		});
+		it("should execute fast", function (done) {
+        	expect(enhancedDate.setDate(new Date("05-23-2015"))).to.be.an.instanceof(Date); 
+        	done();       	
+		});
+	});
+
+	describe("getDate", function() {
+		before(function() {
+			enhancedDate.setDate(new Date("05-23-2015"));
+		});	
+		it("should return number", function () {
+        	expect(enhancedDate.getDate()).to.be.a("number");        	
+		});
+		it("should execute fast", function (done) {
```

Comment:
These tests aren't asynchronous, so there's no need to verify timing. This test tests the same thing the previous test does. Because this test can return two different kinds of values, there are two clear tests that can be written to validate the two potential return values.